main/zoe/guide/README.md

```diff
@@ -64,176 +63,170 @@ handle the credit cards themselves.
-      want: { Price: null },
-    },
-  });
-```
+The contract first confirms that `issuers` are setup for the `Asset` and `Price` keywords. Those are the two items that will be swapped.
 
-This is pretty similar in format to the `automaticRefund`, but there
-are a few changes. First, in this contract, we use the 
-[`inviteAnOffer` helper function](../api/zoe-helpers.html#zoehelper-inviteanoffer-options) to 
-make an invite that will actually check what was
-escrowed with Zoe to see if it's the kind of offer that we want to
-accept. In this case, we only want to accept offers that have a
-proposal of the form:
-```js
-{ give: { Asset: amount1, want: { Price: amount2 } }
+The following uses the [`assertIssuerKeywords` helper function](../api/zoe-helpers.html#assertissuerkeywords-zcf-keywords). It
+checks properties of the running contract instance's terms. The terms were retrieved via `zcf.getTerms()`
+in the `automaticRefund` contract.
+```javascript
+const start = zcf => {
+  assertIssuerKeywords(zcf, ['Asset', 'Price']);
 ```
-where `amount1` and `amount2` are amounts with the correct issuers.
 
-Also, this is a swap, so we can't immediately return a payout to the
-user who puts in the first offer; we have to wait for a valid matching
-offer. So, if we get a valid first offer, we create an invite which can 
-be shared with other parties to create a matching offer.
-
-So, how does the matching happen? We can look at another user-facing
-method, `makeMatchingInvite`, and a helper function, `swap`:
+The first handler defined below, `makeMatchingInvitation()` is for the contract instance's creator, and it
+makes the `invitation` for the other party to use. At the end of this section, we'll see how it's incorporated into
+the contract. When the associated `invitation` is used to make an offer, `makeMatchingInvitation()` is invoked
+with the `seat` for that offer. 
 
+This contract uses the
+[`assertProposalShape` helper function](../api/zoe-helpers.html#assertproposalshape-seat-expected) to 
+check that the offer proposes the kind of trade the contract accepts. In this case, offers must
+have a proposal of the form:
 ```js
-const makeMatchingInvite = firstOfferHandle => {
-  const {
-    proposal: { want, give },
-  } = zcf.getOffer(firstOfferHandle);
-  return inviteAnOffer({
-    offerHook: offerHandle => swap(firstOfferHandle, offerHandle),
-    customProperties: {
-      asset: give.Asset,
-      price: want.Price,
-      inviteDesc: 'matchOffer',
-    },
-  });
-};
+{ give: { Asset: amount1 }, want: { Price: amount2 } }
 ```
-In the `makeMatchingInvite` method we call the [`swap` helper function](../api/zoe-helpers.html#zoehelper-swap-keephandle-tryhandle-keephandleinactivemsg), which handles a lot of the logic. The code for `swap` is:
+`amount1` and `amount2` are amounts with the correct issuers for the keywords.
+`assertProposalShape()` then pulls out the elements **tyg todo: Which are the elements? The amounts?** of the 
```

Comment:
Replace with:
The contract then uses `getProposal` to extract the properties of the proposal for later matching.